src/plexil-adapter/OwInterface.cpp

```diff
@@ -450,55 +421,44 @@ double OwInterface::groundPosition () const
   return GroundPosition;
 }
 
-static void guarded_move_callback
-(const ow_lander::GuardedMoveResult::ConstPtr& msg)
+static void guarded_move_done_cb
+(const actionlib::SimpleClientGoalState& state,
+ const ow_lander::GuardedMoveResultConstPtr& result)
 {
-  if (msg->success) {
-    GroundFound = true;
-    string frame = msg->frame;
-    string valid = "base_link";
-    if (frame != valid) {  // the only supported value
-      ROS_ERROR("GuardedMoveResult frame was not %s", valid.c_str());
-      return;
-    }
-    GroundPosition = msg->position.z;
-    publish ("GroundFound", true);
-    publish ("GroundPosition", GroundPosition);
-  }
-  else {
-    ROS_WARN("GuardedMove did not find ground!");
-  }
+  ROS_INFO ("GuardedMove finished in state %s", state.toString().c_str());
+  GroundFound = result->success;
+  GroundPosition = result->final.z;
+  publish ("GroundFound", GroundFound);
+  publish ("GroundPosition", GroundPosition);
 }
 
+//////////////////// General Action support ///////////////////////////////
 
-//////////////////// GuardedMove Action support ////////////////////////////////
+const auto ActionTimeoutSecs   = 180.0; // TODO: make this action-specific
+const auto ActionServerTimeout = 10.0;  // seconds
 
-// At present, this is a prototypical ROS action using a dummy server in this
-// directory, GuardedMoveServer.  It is NOT connected to the testbed.
+//////////////////// ROS Action callbacks - generic //////////////////////
 
-static void guarded_move_done_cb
-(const actionlib::SimpleClientGoalState& state,
- const ow_autonomy::GuardedMoveResultConstPtr& result)
+template<typename T>
+static void action_feedback_cb (const T& feedback)
```

Comment:
Again, it doesn't seem there is a purpose to the template here
```suggestion
static void action_feedback_cb (const FeedbackPtr& feedback)
```

Reply:
Your suggestion doesn't compile, as FeedbackPtr does not name a class.